File: practica n6/prestamo.py
from datetime import datetime, timedelta


class Prestamo:
    def __init__(self, estudiante, libro):
        self.estudiante = estudiante
        self.libro = libro
        self.fecha_prestamo = datetime.now()
        self.fecha_devolucion = self.fecha_prestamo + timedelta(days=7)


    def mostrar_info(self) -> None:
        titulo = self.libro.get_titulo() if hasattr(self.libro, 'get_titulo') else getattr(self.libro, 'titulo', str(self.libro))
        print(f"\nPrestamo de libro: {titulo}")
        if hasattr(self.estudiante, 'mostrar_info'):
            self.estudiante.mostrar_info()
        else:
            print(self.estudiante)
        print("Fecha de prestamo:", self.fecha_prestamo)
        print("Fecha de devolucion:", self.fecha_devolucion)

File: practicas/bibli/prestamo.java
package practicas.bibli;
import java.util.Date;

public class prestamo {
    private Date fechaPrestamo;
    private Date fechaDevolucion;
    private estudiante estudiante; 
    private libro libro; 

    public prestamo(estudiante estudiante, libro libro) {
        this.estudiante = estudiante;
        this.libro = libro;
        this.fechaPrestamo = new Date();

        this.fechaDevolucion = new Date(fechaPrestamo.getTime() + 7L*24*60*60*1000);
    }
    public void mostrarInfo() {
        System.out.println("Prestamo de libro: " + libro.getTitulo());
        estudiante.mostrarInfo();
        System.out.println("Fecha de prestamo: " + fechaPrestamo);
        System.out.println("Fecha de devolucion: " + fechaDevolucion);
    }
}